wgpu-hal/src/vulkan/device.rs

```diff
@@ -1012,10 +1016,53 @@ impl crate::Device<super::Api> for super::Device {
             })
             .collect::<Vec<_>>();
 
+        let dsl_create_flags = if self.shared.private_caps.update_after_bind {
+            vk::DescriptorSetLayoutCreateFlags::UPDATE_AFTER_BIND_POOL_EXT
```

Comment:
we should be more deliberate here, and only enable it if any of the BGL entries exceed the base limits